data/data-pipeline/data_pipeline/tests/sources/test_etl_base.py

```diff
@@ -0,0 +1,244 @@
+        etl.output_df.to_csv(
+            path_or_buf=self._DATA_DIRECTORY_FOR_TEST
+            / self._TRANSFORM_CSV_FILE_NAME,
+            index=False,
+        )
+
+        # After running load, write the results as the "output.csv" in the test
+        # directory.
+        etl.load()
+        copy_data_files(
+            src=etl._get_output_file_path(),
+            dst=self._DATA_DIRECTORY_FOR_TEST / self._OUTPUT_CSV_FILE_NAME,
+        )
+
+    def test_existence_of_test_fixtures_base(self):
+        """Every ETL test should have these two test fixture files.
+
+        Can be run without modification for all child classes.
+        """
+        assert (
+            self._DATA_DIRECTORY_FOR_TEST / self._TRANSFORM_CSV_FILE_NAME
+        ).exists()
+
+        assert (
+            self._DATA_DIRECTORY_FOR_TEST / self._OUTPUT_CSV_FILE_NAME
+        ).exists()
+
+    def test_init_base(self, mock_etl, mock_paths):
+        """Test whether class has appropriate parameters set.
+
+        Can be run without modification for all child classes.
+        """
+        # Setup
+        etl = self._get_instance_of_etl_class()
+        data_path, tmp_path = mock_paths
+
+        assert etl.DATA_PATH == data_path
+        assert etl.TMP_PATH == tmp_path
+
+        # Also make sure all parameters that need to be non-null are non-null
+        assert etl.NAME is not None
+        assert etl.LAST_UPDATED_YEAR is not None
+        assert etl.SOURCE_URL is not None
+        assert etl.GEO_LEVEL is not None
+        assert etl.COLUMNS_TO_KEEP is not None
+        assert len(etl.COLUMNS_TO_KEEP) > 0
+
+        # Check certain parameters are set.
+        assert etl.EXPECTED_MAX_CENSUS_BLOCK_GROUPS == 250000
+        assert etl.EXPECTED_MAX_CENSUS_TRACTS == 74160
+
+    def test_get_output_file_path_base(self, mock_etl, mock_paths):
+        """Test file path method.
+
+        Can be run without modification for all child classes.
+        """
+        etl = self._get_instance_of_etl_class()
+        data_path, tmp_path = mock_paths
+
+        actual_file_path = etl._get_output_file_path()
+
+        expected_file_path = (
+            data_path
+            / "dataset"
+            / f"{etl.NAME}_{etl.LAST_UPDATED_YEAR}"
+            / "usa.csv"
+        )
+
+        logger.info(f"Expected: {expected_file_path}")
+
+        assert actual_file_path == expected_file_path
+
+    def test_transform_sets_output_df(self, mock_etl, mock_paths):
+        """This test ensures that the transform step sets its results to `output_df`.
+
+        Can be run without modification for all child classes.
+        """
+        etl = self._setup_etl_instance_and_run_extract(
+            mock_etl=mock_etl, mock_paths=mock_paths
+        )
+        etl.transform()
+
+        assert etl.output_df is not None
+
+        # Assert it has some rows
+        assert etl.output_df.shape[0] > 0
+
+        # Check that it has all columns
+        for col in etl.COLUMNS_TO_KEEP:
```

Comment:
I think if there's no uniqueness test for the columns, it might make sense to check shape[1] with the idea that having every column and having every column precisely one time are not the same.

Reply:
I added a uniqueness test on `COLUMNS_TO_KEEP` - let me know if you still think we should update this.

Reply:
I think the uniqueness test is sufficient!